### DISCUSSION TITLE
Gen mask for stations

### DISCUSSION THREAD

**Discussion by Deba57** [dtcenter/METplus#2614] [Configuration] [answered]
Hi,
I am using 5 km model forecast to validate over point locations using Pointstat tool in Metplus. I am wondering whether there is any way to generate mask over individual stations (by providing lat, lon) in GenMask in Metplus and tun that list of stations in Pointstar. Can someone point to an example for such kind. 
Thanks
Deba

**Comment by j-opatz** [comment, marked as answer]
Hello Deba,

And thank you for your question. If your stations have accompanying station IDs, there is a quick and easy way to accomplish your goal using only Point-Stat. To do this, you should utilize the `mask.sid` setting: in the Point-Stat configuration file, you'll find a mask library:

```
mask = {
   grid    = [ "FULL" ];
   poly    = [ ];
   sid     = [ ];
   llpnt   = [ ];
}
```

The `sid` array can accept strings of a filename or comma-separated list of station IDs that will be used to generate a mask. Take a look at [this section of the MET User's Guide](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#mask) to understand how to pass each type of argument to MET.

If you intend to use the PointStat wrapper, the setting would be POINT_STAT_MASK_SID and would accept the same arguments.

If station ID's are not available, there _is_ a way to complete the task you're describing by using the Gen-Vx-Mask tool.

To use individual lat/lon locations ("stations") to generate a mask, you'll need to use the `box` or `circle` types. Both of these types allow a user to input an ASCII file that contains station coordinates to METplus and have it create a perimeter around each station. As you may be able to guess, the perimeter shape and size will be determined by the use of `box` (create a rectangle around each station by specifying how many grid units in width and length you'd like to include) or `circle` (a great-arc distance from each grid point to the stations that can be combined with the `thresh` option to set a radius of the allowed distance).

For a more detailed discussion of these two settings, check out [the MET User's Guide entry](https://met.readthedocs.io/en/latest/Users_Guide/masking.html#types-of-masking-available-in-gen-vx-mask) about the different types of masking, specifically `box` and `circle`.

**Comment by j-opatz** [comment]
Hi Deba,

I wanted to reach out and see if the above suggestion has answered your question. If it has, please mark an answer and I'll go ahead and lock the question from further discussions.

If you still have questions related to this topic or the suggestions didn't quite solve the issue, please reach out and let me know!
